docs/contributing/999-rfcs/2023-10-01-workloads.md

```diff
@@ -0,0 +1,262 @@
+4. [fly.io](https://fly.io)
+5. [ControlPlane](https://github.com/controlplane-com/terraform-provider-cpln/blob/main/docs/resources/workload.md#nestedblock--options)
+
+The default behavior for `tf-aws` should be to create a dedicated EKS cluster for the app (P1) but
+users should be able to point to an existing EKS cluster through platform configuration options
+(P2).
+
+## Autoscaling
+
+> Reqtag: `c:autoscaling`
+
+It should be possible to define autoscaling parameters for a workload. 
+
+Further research is needed to determine the desired degrees of freedom and default behavior. Ideally the default should "just work" and scale up and down elastically based on best practices and common metrics.
+
+
+
+## Preflight object bindings
+
+> Reqtag: `c:bindings`
+
+Containers should be able to interact with Wing preflight objects through their inflight APIs
+(similarly to how inflight closures can invoke inflight APIs of captured preflight objects). Since
+the Winglang compiler cannot infer this automatically, we will need an API for explicitly specifying
+these bindings.
+
+We will also need a way for the code inside the container to be able to connect to the relevant
+inflight client instance and interact with the object.
+
+We can decompose this requirement into four:
+
+1. Call non-Wing services using native clients (P1) (e.g. use the Redis client in Go to interact with a
+   Redis resource). In this case, the code inside the container just uses a native client to
+   interact with the cloud resource but we need a way to give the container the right connection
+   string in order to be able to work across simulator and the cloud (in this case we don't have an
+   abstraction for the resource).
+1. Call Wing SDK clients from within a TypeScript/JavaScript container (P1). This means that the
+   inflight client JavaScript code will need to be injected into the container during build, but the
+   container code can just `import` the `@winglang/sdk` type definitions.
+1. Call any SDK and user-defined Wing inflight API from TypeScript/JavaScript (P2). This means that
+   we will need to inject the inflight client of Wing classes into the container during build and
+   somehow emit typescript type definitions for development (another reason why the Wing compile
+   needs to perform the build).
+1. Call any Wing inflight API from other languages (P3): this is lower priority for now given the
+   technical complexity but it would be healthy to start thinking about the path forward here. This
+   could also be a multi-phased project where we start with something like a sidecar/child process
+   (with generated client bindings) and then compile our inflight clients to WebAssembly when the
+   ecosystem is mature enough (currently there is no AWS SDK for WebAssembly). 
+
+The bindings APIs will need to also allow the user to specify the qualification of the bind (e.g.
+the permission requirements) so that the relevant permissions can be created.
```

Comment:
Probably also needs to handle subnet specifications

Reply:
I think that's more related to the platform implementation, no?

Reply:
I think platforms  are the solution to permissions and networking.

Though my thought process here is if we allow this bind for permissions why not allow easy network configs as well? 

I.E. someone who doesn’t want to write a platform for a small use case and just needs their containers to be added to a security group to access maybe some Redis cluster.

unless I’m thinking about this wrong

Reply:
Perhaps... Let's see how this evolves and iterate.